forte/data/ontology/core.py

```diff
@@ -865,5 +867,167 @@ def index_key(self) -> int:
         return self.tid
 
 
+class Grid:
+    """
+    Regular grid with a grid configuration dependent on the image size.
+
+    We intialize and constrain the grid shape by the image shape during
+    initialization since usually there aren't many different image sizes in one dataset.
```

Comment:
The idea I am trying to convey is that the grid can be totally "free-form" that we don't initialize it with any image sizes and pass the image size directly into the method/operation on the fly. However, since the number of different image shapes are limited, we can bound one grid to one image size, and we can also do the image/grid size check during the initialization.

Reply:
Please reconsider the message from the user perspective, and see if they can understand what you want to do here. The users haven't gone through the whole design phase as you so they don't understand why these considerations matter.

Also, consider whether this message should be placed on the docstring (since this class can be used for and other purposes), or it should be placed in some sort of tutorial.